src/Raven.Server/Documents/Indexes/Persistence/Corax/WriterScopes/EnumerableWriterScope.cs

```diff
@@ -79,15 +82,28 @@ public void Write(int field, BlittableJsonReaderObject reader, ref IndexEntryWri
             _count.Raws++;
         }
 
+        public void Write(int field, CoraxSpatialPointEntry entry, ref IndexEntryWriter entryWriter)
+        {
+            _count.Spatials++;
+            _spatialValues.Add(entry);
+        }
+
         public void Finish(int field, ref IndexEntryWriter entryWriter)
         {
-            if (_count.Raws > 0 && (_count.Longs | _count.Doubles | _count.Strings) != 0)
+            if (_count.Spatials != 0)
+            {
+                if (_count.Spatials == 1)
+                    entryWriter.WriteSpatial(field, _spatialValues[0]);
```

Comment:
Why we need special handling for case count == 1? Why `CollectionsMarshal.AsSpan(_spatialValues)` wouldn't work?